services/data-record-service.js

```diff
@@ -68,66 +71,85 @@ class DataRecordService {
-        const dataRecords = await this.dataRecordsCollection.aggregate(pipeline);
-        const qcResults = await Promise.all(dataRecords.map(async dataRecord => {
-            const latestBatchID = dataRecord.batchIDs?.slice(-1)[0];
-            const latestBatch = (await this.batchCollection.find(latestBatchID)).pop();
-            const severity = dataRecord.status;
-            let description = [];
-            if (severity === VALIDATION_STATUS.ERROR) {
-                description = dataRecord.errors;
-            }
-            if (severity === VALIDATION_STATUS.WARNING) {
-                description = dataRecord.warnings;
-            }
-            return {
-                submissionID: dataRecord.submissionID,
-                nodeType: dataRecord.nodeType,
-                batchID: latestBatchID,
-                displayID: latestBatch?.displayID,
-                nodeID: dataRecord.nodeID,
-                CRDC_ID: dataRecord._id,
-                severity: severity,
-                uploadedDate: latestBatch?.updatedAt,
-                description: description
-            };
-        }));
-        if (!!orderBy){
-            const defaultSort = "uploadedDate";
-            const sort = getSortDirection(sortDirection);
-            qcResults.sort((a, b) => {
-                let propA = a[orderBy] || a[defaultSort];
-                let propB = b[orderBy] || a[defaultSort];
-                if (propA > propB){
-                    return sort;
+        if (severities === ERROR){
+            severities = [ERROR];
+        }
+        else if (severities === WARNING){
+            severities = [WARNING];
+        }
+        else {
+            severities = [ERROR, WARNING];
+        }
+        pipeline.push({
+            $match: {
+                severity: {
+                    $in: severities
                 }
-                if (propA < propB){
-                    return sort * -1;
+            }
+        })
+        if (!!nodeTypes && nodeTypes.length > 0) {
+            pipeline.push({
+               $match: {
+                   nodeType: {
+                       $in: nodeTypes
+                   }
+               }
+            });
+        }
+        if (!!batchIDs && batchIDs.length > 0) {
+            pipeline.push({
+                $match: {
+                    batchID: {
+                        $in: batchIDs
+                    }
                 }
-                return 0;
             });
         }
+        pipeline.push({
+            $sort: {
+                [orderBy]: getSortDirection(sortDirection)
+            }
+        });
+        pipeline.push({
+            $skip: offset
+        });
+        pipeline.push({
+            $limit: first
+        });
+        const dataRecords = await this.dataRecordsCollection.aggregate(pipeline);
         return {
-            total: qcResults.length,
-            results:qcResults.slice(offset, offset+first)
+            total: dataRecords.length,
```

Comment:
I've faced the same issue. When the **limit** is set, the total count of elements is not the total number of elements in db.

Reply:
I've updated the pipeline to include the correct total